examples/compgcn/comgcn_trainer.py

```diff
@@ -0,0 +1,177 @@
+        metrics: tensorlayerx.metrics
+
+    Returns:
+        rst
+    """
+
+    metrics.update(logits, y)
+    rst = metrics.result()
+    metrics.reset()
+    return rst
+
+def main(args):
+    # load cora dataset
+    if str.lower(args.dataset) not in ['aifb', 'mutag', 'bgs', 'am']:
+        raise ValueError('Unknown dataset: {}'.format(args.dataset))
+    path = osp.join(osp.dirname(osp.realpath(__file__)), '../Entities')
+    dataset = Entities(path, args.dataset)
+    graph = dataset[0]
+
+    graph.numpy()
+
+
+    graph.tensor()
+
+    train_y = graph.train_y
+    test_y = graph.test_y
+    edge_index = graph.edge_index
+    edge_type = graph.edge_type
+
+
+    net = CompGCN(feature_dim=32,
+               hidden_dim=args.hidden_dim,
+               num_class=int(dataset.num_classes),
+               num_relations=dataset.num_relations,
+               num_entity = dataset[0].num_nodes,
+               op = args.op,
+               name="CompGCN")
+
+    optimizer = tlx.optimizers.Adam(lr=args.lr, weight_decay=args.l2_coef)
+    metrics = tlx.metrics.Accuracy()
+    train_weights = net.trainable_weights
+
+    loss_func = SemiSpvzLoss(net, tlx.losses.softmax_cross_entropy_with_logits)
+    train_one_step = TrainOneStep(loss_func, optimizer, train_weights)
+
+    #Preprocess Entity Index and Relation Index,
+    #the first half of edge_index and edge_type are output
+    #the second half of edge_index and edge_type are inverse
+    edge_type = tlx.ops.convert_to_numpy(edge_type)
+    edge_index = tlx.ops.convert_to_numpy(edge_index)
+    edge_in_index = [[], []]
+    edge_out_index = [[], []]
+    edge_in_type = []
+    edge_out_type = []
+    for i in range(0, edge_type.shape[0]):
+        if edge_type[i] % 2 == 0:
+            edge_out_index[0].append(edge_index[0][i])
+            edge_out_index[1].append(edge_index[1][i])
+            edge_out_type.append(edge_type[i])
+        else:
+            edge_in_index[0].append(edge_index[0][i])
+            edge_in_index[1].append(edge_index[1][i])
+            edge_in_type.append(edge_type[i])
+
+    edge_index = [[], []]
+
+    edge_index[0] = edge_in_index[0] + edge_out_index[0]
+    edge_index[1] = edge_in_index[1] + edge_out_index[1]
+    edge_type = edge_in_type + edge_out_type
+
+    edge_index = tlx.ops.convert_to_tensor(edge_index)
+    edge_type = tlx.ops.convert_to_tensor(edge_type)
+
+
+    data = {
+        'edge_index': edge_index,
+        'edge_type': edge_type,
+        'train_idx': graph.train_idx,
+        'test_idx': graph.test_idx,
+        'train_y': graph.train_y,
+        'test_y': graph.test_y,
+    }
+
+    best_val_acc = 0
+    for epoch in range(args.n_epoch):
+        net.set_train()
+        train_loss = train_one_step(data, train_y)
+        net.set_eval()
+        logits = net(data['edge_index'], data['edge_type'])
+        val_logits = tlx.gather(logits, data['test_idx'])
+        val_acc = calculate_acc(val_logits, data['test_y'], metrics)
```

Comment:
Why the variable is valid accuracy but we use ``test`` in the input parameters.

Reply:
hhhh, the paradigm of train and val is what I copy from RGCN(ggl), maybe it doesn't have valid dataset, what about we ask RGCN together.